Makefile

```diff
@@ -127,11 +171,9 @@ bazel-generate:
 # find or download controller-gen
 # download controller-gen if necessary
 controller-gen:
-ifeq (, $(shell which controller-gen))
+ifeq (, $(CONTROLLER_GEN))
 	@{ \
 	go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.10.0 ;\
 	}
-CONTROLLER_GEN=$(GOBIN)/controller-gen
-else
```

Comment:
I don't think you can skip the `if-else` here. GOBIN path is not in PATH so you need to hard set the `CONTROLLER_GEN` when you install it with `go install`.

Related to this, please have a look at the PR here: https://github.com/kubev2v/forklift-controller/pull/463. One of the thing it does is it fixes the `controller-gen` target. What we have here is essentially broken and does not work people would think (hint: `controller-gen` is empty target and does not do anything). I closed it because with our transition to Basel I believed it won't be needed and we will drop the Makefile completely. But since it is not the case the PR is still relevant and should be applied.

Reply:
Again, you are right, I took the following approach: use `$(GOBIN)/controller-gen` as the default command, and if not changed by setting the `CONTROLLER_GEN` variable, make sure the required version is installed. what do you think?

Reply:
I like the approach. Now fixing the target will be even easier since we won't need `if-else` anymore (make will figure it on it's own). Something like this should be enough:

```
.PHONY: controller-gen
controller-gen: $(CONTROLLER_GEN)
$(DEFAULT_CONTROLLER_GEN):
	go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.10.0
```

Reply:
Great! I just updated the code to reflect that.